app/cdap/components/WranglerGrid/SelectColumnPanel/DataTable/TableRow/index.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import InputWidget from 'components/WranglerGrid/SelectColumnPanel/DataTable/InputWidgets';
+import DataQualityCircularProgressBar from 'components/common/DataQualityCircularProgressBar';
+import { ITableRowProps } from 'components/WranglerGrid/SelectColumnPanel/DataTable/types';
+import { TableCellText } from 'components/common/TypographyText';
+import { TableRow, TableCell } from '@material-ui/core';
+import { grey } from '@material-ui/core/colors';
+import styled from 'styled-components';
+
+const SelectColumnTableRow = styled(TableRow)`
+  font-weight: 400;
+  font-size: 16px;
+  line-height: 150%;
+  letter-spacing: 0.15px;
+  color: ${grey[700]};
+  display: grid;
+  grid-template-columns: 8% 45% 45%;
+  align-items: center;
+  height: 100%;
+`;
+
+const SelectColumnTableBodyCell = styled(TableCell)`
+  &.MuiTableCell-body {
+    font-weight: 400;
+    font-size: 16px;
+    line-height: 150%;
+    letter-spacing: 0.15px;
+    color: ${grey[700]};
+    padding: 5px;
+    height: 64px;
+  }
+`;
+
+const SelectColumnInputTableBodyCell = styled(SelectColumnTableBodyCell)`
+  &.MuiTableCell-body {
+    padding-left: 11px;
+  }
+`;
+
+export default function({
+  onSingleSelection,
+  selectedColumns,
+  dataQualityValue,
+  isSingleSelection,
+  handleDisableCheckbox,
+  onMultipleSelection,
+  columnIndex,
+  columnDetail,
+}: ITableRowProps) {
+  return (
+    <SelectColumnTableRow key={`column-${columnIndex}`}>
+      <SelectColumnInputTableBodyCell>
+        <InputWidget
+          isSingleSelection={isSingleSelection}
+          selectedColumns={selectedColumns}
+          onSingleSelection={onSingleSelection}
+          columnDetail={columnDetail}
+          handleDisableCheckbox={handleDisableCheckbox}
+          onMultipleSelection={onMultipleSelection}
+          columnIndex={columnIndex}
+        />
+      </SelectColumnInputTableBodyCell>
+      <SelectColumnTableBodyCell>
+        <TableCellText component="div">{columnDetail.label}</TableCellText>
+        <TableCellText component="div">{columnDetail.type}</TableCellText>
+      </SelectColumnTableBodyCell>
+      <SelectColumnTableBodyCell>
+        {dataQualityValue?.length && (
+          <DataQualityCircularProgressBar
+            dataQualityPercentValue={Number(dataQualityValue[columnIndex]?.value)}
```

Comment:
what if the value is undefined, what is Number(undefined)? NaN
so plz make sure whether we need this check here
or can we remove the optional chaining safely if the columnIndex has value for sure

please provide proper explanation

Reply:
It's going to have value for sure, we are calculating it on the basis of api data, and if don't get data from api then also this case is handled in its parent component